src/Oscoin/Data/Ledger.hs

```diff
@@ -225,6 +259,28 @@ deriving instance (Eq (Crypto.Hash c), Eq (AccountId c))     => Eq   (Dependency
 deriving instance (Ord (Crypto.Hash c), Ord (AccountId c))   => Ord  (DependencyUpdate c)
 deriving instance (Show (Crypto.Hash c), Show (AccountId c)) => Show (DependencyUpdate c)
 
+instance ( Serialise (Crypto.Hash c)
+         , Serialise (AccountId c)
+         , Serialise (Signature c)
+         ) => Serialise (DependencyUpdate c)
+  where
+    encode (Depend proj hsh) =
+           CBOR.encodeListLen 3
+        <> CBOR.encodeWord 0
+        <> CBOR.encode proj
+        <> CBOR.encode hsh
+    encode (Undepend proj) =
+           CBOR.encodeListLen 2
+        <> CBOR.encodeWord 1
+        <> CBOR.encode proj
+
+    decode = do
+        pre <- liftA2 (,) CBOR.decodeListLen CBOR.decodeWord
+        case pre of
+            (3, 0) -> Depend   <$> CBOR.decode <*> CBOR.decode
+            (2, 1) -> Undepend <$> CBOR.decode
```

Comment:
Small nit -- why not pattern match first only on the tag (which uniquely identifies the constructor) and then check that the length is the requested one? I guess now it doesn't matter, but it would give you more precise errors in case you successfully decoded the tag but the length mismatched -- if anything, you will know which constructor failed to decode vs now where the failure is quite opaque. What do you think?

Reply:
Hmm I don't think it matters too much as long as we only have this implementation, since the roundtrip is tested. Perhaps in the future it would be helpful to know exactly what was wrong?

Reply:
Yeah, this is more for the future -- the good news is that we can change it later as it won't alter the structure of the payload, only the way we decode it.